HIS/his_checker.py

```diff
@@ -0,0 +1,177 @@
+        num_statements = int(fields[2])
+        if num_statements > statements_treshold:
+            function = fields[5].split(':')
+            print(str(function[1]) + " exceeded the # of statements allowed with "
+                 + fields[2] + " statements (max. " + str(statements_treshold) + 
+                  ")" + " at " + str(function[0]))
+            sys.exit(-1)
+
+def process_goto(file):
+
+    """Process number of go to statements"""
+
+    qmcalc = "qmcalc "
+    lines = os.popen(qmcalc + str(file))
+    lines = lines.read()
+    sline = lines.split('\n')
+
+    for num in range(len(sline)-1):
+        fields = sline[num].split('\t')
+        if(int(fields[18]) > goto_treshold):
+            print("Illegal goto expression in a function in" + file)
+            sys.exit(-1)
+
+def process_calls(file):
+
+    """Process number of calls inside a function"""
+
+    cflow = "cflow -l --depth=2 "
+    if file.endswith(".c"):
+        lines = os.popen(cflow + str(file))
+        lines = lines.read()
+        sline = lines.split('\n')
+        file_path = []
+        call_counter = 0
+
+        for num in range(len(sline)-1):
+            fields = sline[num].split('\t')
+            fields = fields[0].split('}')
+            fields[0] = fields[0].lstrip("{")
+            fields[0] = fields[0].lstrip()
+            fields[1] = fields[1].lstrip()
+
+            if (int(fields[0]) == 0 and call_counter > call_treshold):
+                print("With " + str(call_counter) + " calls (of a maximum "
+                        + str(call_treshold) + "): " + parent_func)
+                sys.exit(-1)
+
+            if (int(fields[0]) == 0): 
+                parent_func = fields[1]
+                call_counter = 0
+
+            if (int(fields[0]) > 0):
+                call_counter = call_counter + 1
+
+def process_callees(file):
+
+    """Process number of times a function is called in a file"""
+
+    cflow = "cflow -r -l --depth=2 "
+    if file.endswith(".c"):
+        lines = os.popen(cflow + str(file))
+        lines = lines.read()
+        sline = lines.split('\n')
+        file_path = []
+        callee_counter = 0
+
+        for num in range(len(sline)-1):
+            fields = sline[num].split('\t')
+            fields = fields[0].split('}')
+            fields[0] = fields[0].lstrip("{")
+            fields[0] = fields[0].lstrip()
+            fields[1] = fields[1].lstrip()
+
+            if (int(fields[0]) == 0 and callee_counter > callee_treshold):
+                print(parent_func + " is being called " + str(callee_counter) +
+                      " times (from a maximum of " + str(callee_treshold) + 
+                      ")" + " in " + file)
+                sys.exit(-1)
+
+            if (int(fields[0]) == 0): 
+                parent_func = fields[1]
+                callee_counter = 0
+
+            if (int(fields[0]) > 0):
+                callee_counter = callee_counter + 1
+
+if __name__ == "__main__" :
```

Comment:
No space allowed before :
```suggestion
if __name__ == "__main__":
```

Reply:
Removed space